### PR TITLE
Feature/next

### PR DESCRIPTION
**Pull request by mjkoo**
Addresses #8 , #9  and partially #7 , polyswarmd support coming

### CODE REVIEW

File: src/apikeymgmt/__init__.py [hunk 1]
```diff
@@ -94,136 +222,188 @@ def after_request(response):
...
+
+        communities = Community.query.filter(db.or_(*[Community.name == name for name in communities])).all()
+
+        api_key = ApiKey(os.urandom(16).hex(), communities)
+
+        user.api_keys.append(api_key)
         db.session.commit()
-    except:
+    except Exception:
+        # Keys are random, but there's a chance of collision
```
Comment (mrtizmoatwork) [inline src/apikeymgmt/__init__.py]: if there is a chance of collision, can the code check for the key's existence before you create a new ApiKey?

Comment (mjkoo) [reply src/apikeymgmt/__init__.py]: Better to let the db handle it via unique constraint and catch the exception vs having a race between the look up and insert imo

Comment (mjkoo) [reply src/apikeymgmt/__init__.py]: Re edit: would rather the client retry than loop in the request handler

Comment (mrtizmoatwork) [reply src/apikeymgmt/__init__.py]: cool

**Pull request merged**